src/include/miopen/op_kernel_args.hpp

```diff
@@ -4,8 +4,11 @@
 #include <type_traits>
 #include <cstdint>
 #include <half.hpp>
-
+#if HIP_PACKAGE_VERSION_FLAT > 5001999999ULL
+#include <miopen/config.h>
```

Comment:
For reviewers' information:
This is puzzling workaround but only reproducible on ROCm 5.1.x: even with config.h included build will fail and complain the same thing about noinline.

Reply:
@junliume How can I reproduce the issue? Tried with ROCm 5.1 (miopen-image built from the scratch using our Dockerfile), but there is no problems with original code (even if config.h is not included). Either I am doing something wrongly or this "puzzling W/A" is not necessary.

Reply:
@atamazov  it only appears in our CI and I have had a hard time trying to reproduce it too.
Here is the log
[before_57340fff.log](https://github.com/ROCmSoftwarePlatform/MIOpen/files/8834720/before_57340fff.log)

Reply:
@junliume Thanks, now I see how to reproduce it (-k analyze)

Reply:
@junliume The reason is as follows. The fix in boost works at compile time, when include/boost/config/detail/suffix.hpp is first included. HIP_VERSION must be defined at the moment of first inclusion of that header. Error message says:
```
[ 47%] Building CXX object speedtests/CMakeFiles/speedtest_sequences.dir/sequences.cpp.o
In file included from /var/jenkins/workspace/MLLibs_MIOpen_boost_update/speedtests/sequences.cpp:4:
In file included from /var/jenkins/workspace/MLLibs_MIOpen_boost_update/speedtests/../test/driver.hpp:28:
...
In file included from /var/jenkins/workspace/MLLibs_MIOpen_boost_update/src/include/miopen/op_kernel_args.hpp:8:
In file included from /usr/local/include/boost/container/small_vector.hpp:27:
/usr/local/include/boost/container/vector.hpp:2753:4: error: use of undeclared identifier 'noinline'; did you mean 'inline'?
   BOOST_CONTAINER_NOINLINE iterator priv_insert_forward_range_no_capacity
   ^
```
We can see that `sequences.cpp` includes `driver.hpp`, then `driver.hpp` includes `get_handle.hpp` and so on, and then build failure happens somewhere in depths (`op_kernel_args.hpp`). However `sequences.cpp` includes some other files **_before_** `driver.hpp`, specifically, `miopen/sequences.hpp` and `miopen/rank.hpp`. 

`miopen/sequences.hpp` includes some boost beaders (`boost/range/algorithm/find.hpp`). At that moment config.h is not included yet, and `BOOST_NOINLINE` is defined incorrectly. I will provide a fix.

Reply:
@junliume Done in #1581